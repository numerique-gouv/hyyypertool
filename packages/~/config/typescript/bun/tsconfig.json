{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "../base/tsconfig.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
