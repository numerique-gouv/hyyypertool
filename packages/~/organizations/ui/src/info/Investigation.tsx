//

import { button } from "@~/app.ui/button";
import { hx_urls } from "@~/app.urls";

import type { Organization } from "@~/organizations.lib/entities/Organization";
import type { JSX } from "hono/jsx";

//

type Props = JSX.IntrinsicElements["section"] & {
  organization: Pick<Organization, "cached_code_postal" | "siret">;
};

//

export async function Investigation(props: Props) {
  const { organization } = props;
  const hx_organizations_leaders_props =
    await hx_urls.organizations.leaders.$get({
      query: { siret: organization.siret },
    });

  const button_classes = button({
    class: "mr-2 bg-white",
    size: "sm",
    type: "tertiary",
  });

  return (
    <ul class="mt-5 w-full list-none bg-[#F6F6F6] p-3 [&_li]:inline-block">
      <li>
        <a
          href={`https://lannuaire.service-public.fr/recherche?where=${organization.cached_code_postal}&whoWhat=Mairie`}
          class={button_classes}
          rel="noopener noreferrer"
          target="_blank"
        >
          Chercher la mairie associée
        </a>
      </li>
      <li>
        <a
          href={`https://lannuaire.service-public.fr/recherche?where=${organization.cached_code_postal}`}
          class={button_classes}
          rel="noopener noreferrer"
          target="_blank"
        >
          Chercher les services publics associés
        </a>
      </li>
      <li>
        <a class="" {...hx_organizations_leaders_props} hx-trigger="load">
          <i class="text-center">Recherche des dirigeants...</i>
        </a>
      </li>
    </ul>
  );
}
