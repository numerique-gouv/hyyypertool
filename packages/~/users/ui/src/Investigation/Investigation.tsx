//

import { z_email_domain } from "@~/app.core/schema/z_email_domain";
import { button } from "@~/app.ui/button";
import { GoogleSearchButton } from "@~/app.ui/button/components/search";
import type { Organization } from "@~/organizations.lib/entities/Organization";
import type { User } from "@~/users.lib/entities/User";

//

type InvestigationProps = {
  user: Pick<User, "email">;
  organization: Pick<Organization, "cached_libelle">;
};

//

export function Investigation(props: InvestigationProps) {
  const { user, organization } = props;

  const domain = z_email_domain.parse(user.email, { path: ["user.email"] });

  return (
    <section {...props}>
      <h4>üïµÔ∏è Enqu√™te sur ce profile</h4>

      <ul class="list-none pl-0">
        <li>
          <GoogleSearchButton
            class={button({ size: "sm", type: "tertiary" })}
            query={user.email}
          >
            R√©sultats Google pour cet email
          </GoogleSearchButton>
        </li>
        <li>
          <GoogleSearchButton
            class={button({ size: "sm", type: "tertiary" })}
            query={domain}
          >
            R√©sultats Google pour ce nom de domaine
          </GoogleSearchButton>
        </li>
        <li>
          <GoogleSearchButton
            class={button({ size: "sm", type: "tertiary" })}
            query={`${organization.cached_libelle} ${domain}`}
          >
            R√©sultats Google pour le nom de l'organisation et le nom de domaine
          </GoogleSearchButton>
        </li>
      </ul>
    </section>
  );
}
