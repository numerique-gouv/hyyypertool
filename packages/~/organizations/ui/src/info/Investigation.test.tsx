//

import { expect, test } from "bun:test";
import { Hono } from "hono";
import { jsxRenderer } from "hono/jsx-renderer";
import { format } from "prettier";
import { Investigation } from "./Investigation";

test("render investigation section", async () => {
  const response = await new Hono()
    .get("/", jsxRenderer(), ({ render }) =>
      render(
        <Investigation
          organization={{
            cached_code_postal: "75015",
            siret: "12345678901234",
          }}
        />,
      ),
    )
    .request("/");
  expect(response.status).toBe(200);
  expect(
    await format(await response.text(), { parser: "html" }),
  ).toMatchSnapshot();
});
