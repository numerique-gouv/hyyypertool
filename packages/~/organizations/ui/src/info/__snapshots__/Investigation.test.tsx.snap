// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`render investigation section 1`] = `
"<!DOCTYPE html>
<section>
  <h4>🕵️ Enquête sur cette organisation</h4>
  <ul class="list-none pl-0">
    <li>
      <a
        href="https://lannuaire.service-public.fr/recherche?where=75015&amp;whoWhat=Mairie"
        class="fr-btn fr-btn--sm fr-btn--tertiary"
        rel="noopener noreferrer"
        target="_blank"
        >Mairie sur Annuaire Service Public</a
      >
    </li>
    <li>
      <a
        href="https://lannuaire.service-public.fr/recherche?where=75015"
        class="fr-btn fr-btn--sm fr-btn--tertiary"
        rel="noopener noreferrer"
        target="_blank"
        >Service sur Annuaire Service Public</a
      >
    </li>
    <li>
      <a
        href="https://data.education.gouv.fr/api/v2/catalog/datasets/fr-en-annuaire-education/records?where=siren_siret%3D12345678901234"
        class="fr-btn fr-btn--sm fr-btn--tertiary"
        rel="noopener noreferrer"
        target="_blank"
        >Établissement sur l&#39;annuaire Éducation Nationale</a
      >
    </li>
  </ul>
</section>
"
`;
